.github/scripts/rust/Makefile.toml

```diff
@@ -0,0 +1,157 @@
+[config.modify_core_tasks]
+private = true
+namespace = "default"
+
+[env]
+CARGO_MAKE_EXTEND_WORKSPACE_MAKEFILE = true
+CARGO_HACK_FLAGS = "--feature-powerset --exclude-features default"
+
+[env.production]
+CARGO_MAKE_CARGO_PROFILE = "release"
+
+
+[tasks.default]
+extend = "build"
+category = ""
+
+[tasks.all]
+description = "Builds the project, checks lints, and runs the tests. Additional parameters are passed to `build` and `test`."
+run_task = { name = ["build", "lint", "test"] }
+
+
+################################################################################
+## Build                                                                      ##
+################################################################################
+[tasks.build]
+description = "Builds the crate"
+category = "Build"
+workspace = false
+command = "cargo"
+args = ["build", "--profile", "${CARGO_MAKE_CARGO_PROFILE}", "${@}"]
+
+
+################################################################################
+## Run                                                                        ##
+################################################################################
+[tasks.run]
+description = "Builds the binary and runs it"
+category = "Run"
+workspace = false
+command = "cargo"
+args = ["run", "--profile", "${CARGO_MAKE_CARGO_PROFILE}", "${@}"]
+
+
+################################################################################
+## Lints                                                                      ##
+################################################################################
+[tasks.lint]
+description = "Runs all lints"
+category = "Lint"
+run_task = { name = ["format-task-ci", "clippy-task-ci", "doc-task-ci-public", "doc-task-ci-private"] }
```

Comment:
I'm a little confused by the `-ci` stuff, I don't find this particularly intuitive. If they're only intended to be ran on `CI` then why are they seemingly ran here on a plain `lint` task. If they're not only meant to be ran on CI then why are they named with `-ci`?